### ISSUE TITLE
ThreadHandle isn't @resource, so an unjoined Thread handle compiles

### ISSUE THREAD

**Issue by dritory** [rask-lang/rask#1360]
`stdlib/thread.rk` says a `ThreadHandle` "must be joined or detached", and `specs/concurrency/README.md`'s spawn table lists "Unused handle: compile error" for all three spawn forms. But `ThreadHandle` isn't `@resource`, so nothing enforces it:

```rask
func main() {
    let t = Thread.spawn(|| { return 5 })
}   // compiles; the handle is never joined or detached
```

`TaskHandle` has been `@resource` since #797.

## Why it wasn't just flipped

I tried it while fixing a double free (`ensure t.detach()` running after `t.join()`, fixed separately in the PR for #1354: MIR took the handle's prefix from the call's shape). The corpus fallout:

- `examples/16_concurrency_basics.rk` collects handles in a `Vec` and joins them in a loop. That becomes E0820, because a `Vec` can't hold a linear value. Tasks had the same problem, and `TaskGroup` is the answer there (#891).
- Every multi-spawn needs an `ensure h.detach()` per handle when anything stands between spawn and join (E0882). That's the same cost `TaskHandle` programs pay today.

So making it linear needs a thread-group equivalent, or `TaskGroup` generalised over both handle kinds, before the example can be kept. That's a design choice, so I'm filing it rather than making it.

Options:
1. `@resource ThreadHandle` plus a `ThreadGroup` (or one group type for both handle kinds). Consistent with tasks and the spec's table.
2. Leave it non-linear and fix the spec and docstring to say a dropped Thread handle is detached. Cheaper, but a second rule for the same shape.
